src/dune_pkg/rev_store.ml

```diff
@@ -157,28 +183,29 @@ let run_with_exit_code { dir; _ } ~allow_codes ~display args =
     Process.run ~dir ~display ~stdout_to ~stderr_to ~env failure_mode git args
   in
   if allow_codes exit_code
-  then exit_code
-  else git_code_error ~dir ~args ~exit_code ~output:[]
+  then Ok exit_code
+  else Error { Git_error.dir; args; exit_code; output = [] }
 ;;
 
 let run t ~display args =
-  run_with_exit_code t ~allow_codes:(Int.equal 0) ~display args >>| ignore
+  run_with_exit_code t ~allow_codes:(Int.equal 0) ~display args
+  >>| Result.map ~f:(fun (_ : int) -> ())
```

Comment:
I know there is some use in annotating the expected type to be ignored but it's also adding a lot of noise, so maybe

```suggestion
  >>| Result.map ~f:ignore
```

Reply:
It's pretty useful to know what's being ignored. I think the noise makes it worth it here. You could also do `Result.map ~f:(ignore : int -> unit)` I suppose

Reply:
Updated to use the `ignore` function as I think it's clearer.